minitask2/akhmad/behaviour.py

```diff
@@ -0,0 +1,196 @@
+        rospy.init_node('behaviour')
+
+        # Subscribers
+        self.laser_sub = rospy.Subscriber('/scan', LaserScan, self.scan_callback)
+        self.odom_sub = rospy.Subscriber('/odom', Odometry, self.odom_callback)
+
+        # Publisher
+        self.cmd_vel_pub = rospy.Publisher('/cmd_vel', Twist, queue_size=3)
+
+        self.rate = rospy.Rate(10)
+        self.current_pose_theta = None
+        self.current_pose_x = None
+        self.current_pose_y = None
+        rospy.spin()
+
+    def publish_command(self, msg):
+        self.cmd_vel_pub.publish(msg)
+
+    def turn_left(self):
+        global continue_left, continue_right
+        continue_right = False
+        continue_left = True
+        self.publish_command(left_turn_cmd)
+        print("Avoiding obstacle by turning left")
+
+    def turn_right(self):
+        global continue_left, continue_right
+        continue_right = True
+        continue_left = False
+        self.publish_command(right_turn_cmd)
+        print("Avoiding obstacle by turning right")
+
+    def right_hand_wall_follow(self):
+        global is_wall_found
+
+        if not is_wall_found:
+            if all(i >= 0.5 for i in front):
+                self.publish_command(move_cmd)
+            else:
+                is_wall_found = True
+                self.publish_command(Twist())
+        else:
+            if min(right) <= 0.5 and min(front) <= 0.5:
+                self.publish_command(left_turn_cmd)
+                print("Obstacles on front and right - turning left")
+            elif min(right) <= 0.5 and min(front) >= 0.5:
+                self.publish_command(move_cmd)
+                print("No obstacles in front but on right - move forward")
+            elif min(right) >= 0.5 and min(front) <= 0.5:
+                self.publish_command(left_turn_cmd)
+                print("No obstacles on right but in front - turning left")
+            elif min(right) >= 0.5 and min(front) >= 0.5:
+                if min(right_front) <= 0.5:
+                    self.publish_command(right_turn_cmd)
+                else:
+                    self.publish_command(move_cmd)
+                print("No obstacles on right and in front - sliding right")
+
+    def obstacle_avoidance(self):
+        # continue_right and continue_left is used to
+        # keep turning in same direction as robot started
+        # otherwise it might get stuck in corners
+        if continue_right:
+            self.turn_right()
+        elif continue_left:
+            self.turn_left()
+        elif all(i >= 0.5 for i in left_front):
+            self.turn_left()
+        elif all(i >= 0.5 for i in right_front):
+            self.turn_right()
+        elif all(i >= 0.5 for i in left_back):
+            self.turn_left()
+        elif all(i >= 0.5 for i in right_back):
+            self.turn_right()
+        else:
+            self.turn_left()
+
+    def random_walk(self):
+        rospy.loginfo("Random walk")
```

Comment:
You've mixed rospy logging with standard print statements. I think `rospy.loginfo` is defo better, but we should really commit to one or the other.

Reply:
I agree. It is better to use `rospy.loginfo` everywhere.